boa/src/builtins/map/mod.rs

```diff
@@ -419,79 +441,62 @@ impl Map {
         args: &[JsValue],
         context: &mut Context,
     ) -> JsResult<JsValue> {
-        if args.is_empty() {
-            return Err(JsValue::new("Missing argument for Map.prototype.forEach"));
-        }
-
-        let callback_arg = &args[0];
+        // 1. Let M be the this value.
+        // 2. Perform ? RequireInternalSlot(M, [[MapData]]).
+        let map = match this {
+            JsValue::Object(obj) if obj.is_map() => obj,
+            _ => return context.throw_type_error("`this` is not a Map"),
+        };
 
-        if !callback_arg.is_function() {
-            let name = callback_arg.to_string(context)?;
-            return context.throw_type_error(format!("{} is not a function", name));
-        }
+        // 3. If IsCallable(callbackfn) is false, throw a TypeError exception.
+        let callback = match args.get_or_undefined(0) {
+            JsValue::Object(obj) if obj.is_function() => obj,
```

Comment:
We should use the `is_callable` function here. At the moment it only checks if the object is a function, that may change in the future.
```suggestion
            JsValue::Object(obj) if obj.is_callable() => obj,
```